activerecord/lib/active_record/relation/calculations.rb

```diff
@@ -158,7 +158,23 @@ def calculate(operation, column_name)
     # See also #ids.
     #
     def pluck(*column_names)
-      if loaded? && (column_names.map(&:to_s) - @klass.attribute_names - @klass.attribute_aliases.keys).empty?
+      if loaded? && @klass.effective_column_name?(column_names)
+        return records.pluck(*column_names)
+      end
+
+      if has_include?(column_names.first)
+        construct_relation_for_association_calculations.pluck(*column_names)
+      else
+        if ActiveRecord::Base.guard_unsafe_raw_sql && !@klass.effective_column_name?(column_names)
```

Comment:
I'd consider moving this into a method of it's own `column_valid?` or something like that.